app/models/user.rb

```diff
@@ -49,10 +49,10 @@ def self.admins
     User.where(admin: true)
   end
 
-  def self.from_omniauth(auth)
+  def self.from_omniauth(auth, email)
     # Set login_dot_gov as Provider for legacy TP Devise accounts
     # TODO: Remove once all accounts are migrated/have `provider` and `uid` set
-    @existing_user = User.find_by_email(auth.info.email)
+    @existing_user = User.find_by_email(email)
```

Comment:
find the user by the passed in, selected email, and not the last used email

and below, set the specified email